client/src/components/all-components/SiteAccessibilityButton.js

```diff
@@ -0,0 +1,142 @@
+import React, { useState } from "react";
+import { bindActionCreators } from "redux";
+import { connect } from "react-redux";
+import { actions as appActions } from "../../reducers/lab1/AppReducer";
+import { actions as mainActions } from "../../reducers/MainReducer";
+import { Panel as ColorPickerPanel } from "rc-color-picker";
+import AccessibilityImage from "../../assets/images/accessibility_icon.png";
+import { Collapse, Card, CardHeader, CardBody } from "reactstrap";
+import {
+  changeTSize,
+  setTextColor,
+  setBackgroundColor,
+} from "../footer/edit/editPage";
+
+const mapStateToProps = (state) => {
+  return {
+    // General
+    state: state,
+  };
+};
+
+const mapDispatchToProps = (dispatch) => {
+  return {
+    actions: bindActionCreators({ ...appActions, ...mainActions }, dispatch),
+  };
+};
+
+const SiteAccessibilityButton = () => {
+  const [open, setOpen] = useState(false);
+  const toggleCollapse = (e) => {
+    e.preventDefault();
+    if (e.target === e.currentTarget) {
+      setOpen(!open);
+    }
+  };
+
+  let state = {
+    fontSize: 0,
+    textColor: false,
+    bgColor: false,
+    displayColorPalette: false,
+    backgroundColor: null,
+    color: null,
+  };
+
+  function changeSize(size) {
+    const state_size = state.fontSize;
+    changeTSize(size);
+    state = { ...state, fontSize: state_size + size };
+  }
+
+  function onTextColorChange(obj) {
+    setTextColor(obj.color);
+    state = { ...state, color: obj.color };
+  }
+
+  function onBgColorChange(obj) {
+    setBackgroundColor(obj.color);
+    state = { ...state, backgroundColor: obj.color };
+  }
+
+  return (
+    <button
+      className="tw-flex tw-fixed tw-right-0 tw-bottom-0 tw-m-4
+      tw-bg-primary-yellow tw-border-none tw-rounded-full
+      tw-p-4 tw-z-50 tw-shadow-md
+      hover:tw-bg-[#f6c832] hover:tw-shadow-lg"
+      onClick={(e) => toggleCollapse(e)}
+    >
+      <img
+        className="tw-aspect-square tw-w-12 tw-h-12 tw-pointer-events-none"
+        src={AccessibilityImage}
+      />
+      <Collapse
+        className="tw-absolute tw-bottom-[0%] tw-right-[150%]"
+        isOpen={open}
+      >
+        <Card
+          className="tw-bg-white tw-flex tw-flex-col tw-shadow-lg tw-shadow-labGray tw-w-[20rem]"
+          outline
+          color="light"
+        >
+          <CardHeader className="tw-bg-primary-blue">
+            <h3 className="tw-title-styling-name tw-text-xl tw-font-bold tw-text-white tw-p-6">
+              Accessibility Tools
+            </h3>
+          </CardHeader>
+          <CardBody className="tw-p-2">
+            <p className="tw-text-left tw-body-styling-name tw-px-3">
+              Text Size Adjuster
+            </p>
+            <div className="tw-flex tw-flex-row tw-justify-evenly tw-gap-2 tw-my-2">
+              <button
```

Comment:
I'm not sure if it's me, but I think these buttons aren't keyboard accessible. I pressed tab multiple times and the button wouldn't get highlighted.

Reply:
This is an issue with currently hardcoded text sizes due to tailwind config. We are looking into how we can update this.

Reply:
fixed 👍

Reply:
Can confirm the buttons are keyboard accessible now